packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -350,28 +360,39 @@ export class AzureCommunicationCallAdapter implements CallAdapter {
     });
   }
 
-  private _joinCall(audioOptions: AudioOptions, videoOptions: VideoOptions): Call {
+  private _joinCall(audioOptions: AudioOptions, videoOptions: VideoOptions): CallTypeOf<AgentType> {
     const isTeamsMeeting = 'meetingLink' in this.locator;
     /* @conditional-compile-remove(rooms) */
     const isRoomsCall = 'roomId' in this.locator;
 
+    /* @conditional-compile-remove(teams-identity-support) */
+    if (_isTeamsCallAgent(this.callAgent)) {
+      if (isTeamsMeeting) {
+        return this.callAgent.join(this.locator as TeamsMeetingLinkLocator, {
```

Comment:
Avoid this duplicate block which is the same as below. Instead invert the if:
```ts
if (_isTeamsCallAgent(this.callAgent) && !isTeamsMeeting) {
  throw new Error('Locator not supported by TeamsCallAgent');
}
...
if (isTeamsMeeting) {
  ...
```

Reply:
Noticed there are 2 teams meeting condition here - one for CTE, one for regular ACS. So _isTeamsCallAgent(this.callAgent) still need to be there as a type guard, but I applied if(!isTeamsMeeting) {throw ""} to avoid an else closure :D